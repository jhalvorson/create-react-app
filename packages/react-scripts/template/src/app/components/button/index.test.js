import React from 'react';
import { shallow } from 'enzyme';
import Button from './';

it('renders without crashing', () => {
  shallow(<Button />);
});


it('should render some text', () => {
  const props = {
    onClick: () => {},
  };

  const wrapper = shallow(
    <Button onClick={props.onClick}>Press me!</Button>,
  );
  expect(wrapper).toMatchSnapshot();
});

/**
 * We can pass in click events using simulate
 */
it('renders the children passed in and can be pressed', () => {
  const props = {
    onClick: () => console.log('PRESSED'),
  };

  const wrapper = shallow(<Button onClick={props.onClick}>Press me!</Button>);
  expect(wrapper.prop('children')).toEqual('Press me!');
  wrapper.simulate('click');
});
