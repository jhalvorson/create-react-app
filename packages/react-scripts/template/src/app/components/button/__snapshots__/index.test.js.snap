// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render some text 1`] = `
<button
  className="button button-primary"
  onClick={[Function]}
>
  Press me!
</button>
`;
